aws-lc-rs/src/aead.rs

```diff
@@ -876,4 +1443,365 @@ mod tests {
+                (Ok(_), true) => panic!("expected error for seal_in_place_append_tag"),
+                (Ok(_), false) => {}
+                (Err(_), true) => return,
+                (Err(e), false) => panic!("{e}"),
+            }
+
+            assert_ne!(plaintext, in_out[..plaintext.len()]);
+
+            opening_key
+                .open_in_place(
+                    Nonce::try_assume_unique_for_key(nonce_bytes).unwrap(),
+                    Aad::empty(),
+                    &mut in_out,
+                )
+                .unwrap();
+
+            assert_eq!(plaintext, in_out[..plaintext.len()]);
+        }
+    }
+
+    #[test]
+    fn test_aes_256_tls12() {
+        struct TestCase {
+            nonce: &'static str,
+            expect_err: bool,
+        }
+
+        let cases = [
+            TestCase {
+                nonce: "9fab40177c900aad9fc28cc3",
+                expect_err: false,
+            },
+            TestCase {
+                nonce: "9fab40177c900aad9fc28cc4",
+                expect_err: false,
+            },
+            TestCase {
+                nonce: "9fab40177c900aad9fc28cc2",
+                expect_err: true,
+            },
+        ];
+
+        let key =
+            from_hex("7476349faa770fcb438e4fb82a93a4db6cd7ad32d9d77868d1bb399423b1a1f3").unwrap();
+        let plaintext = from_hex("00112233445566778899aabbccddeeff").unwrap();
+
+        let sealing_key =
+            TLSRecordSealingKey::new(&AES_256_GCM, TLSProtocolId::TLS12, &key).unwrap();
+
+        let opening_key =
+            TLSRecordOpeningKey::new(&AES_256_GCM, TLSProtocolId::TLS12, &key).unwrap();
+
+        assert_eq!(&AES_256_GCM, sealing_key.algorithm());
+        assert_eq!(16, AES_256_GCM.tag_len());
+        assert_eq!(12, AES_256_GCM.nonce_len());
+
+        for case in cases {
+            let mut in_out = Vec::from(plaintext.as_slice());
+
+            let nonce = from_hex(case.nonce).unwrap();
+
+            let nonce_bytes = nonce.as_slice();
+
+            let result = sealing_key.seal_in_place_append_tag(
+                Nonce::try_assume_unique_for_key(nonce_bytes).unwrap(),
+                Aad::empty(),
+                &mut in_out,
+            );
+
+            match (result, case.expect_err) {
+                (Ok(_), true) => panic!("expected error for seal_in_place_append_tag"),
+                (Ok(_), false) => {}
+                (Err(_), true) => return,
+                (Err(e), false) => panic!("{e}"),
+            }
+
+            assert_ne!(plaintext, in_out[..plaintext.len()]);
+
+            opening_key
+                .open_in_place(
+                    Nonce::try_assume_unique_for_key(nonce_bytes).unwrap(),
+                    Aad::empty(),
+                    &mut in_out,
+                )
+                .unwrap();
+
+            assert_eq!(plaintext, in_out[..plaintext.len()]);
+        }
+    }
+
+    #[test]
+    fn test_aes_256_tls13() {
```

Comment:
There seems to be some shared code among the test functions, would it be worth abstracting them out into another function for reuse?